rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js

```diff
@@ -1293,19 +1297,58 @@ function buildComplianceBar(compliance) {
       text.push("Unknown reports: "+unknown+"%");
     }
     var value = Number((unexpected).toFixed(0));
-    content.append('<div class="progress-bar progress-bar-unknown" style="width:'+unexpected+'%" title="'+text.join("\n")+'">'+value+'%</div>');
+    content.append('<div class="progress-bar progress-bar-unknown" style="width:'+widthArr[4]+'" title="'+text.join("\n")+'">'+value+'%</div>');
   }
 
-  var error = compliance[3]
   if(error != 0) {
     var value = Number((error).toFixed(0));
-    content.append('<div class="progress-bar progress-bar-error" style="width:'+error+'%" title="Error: '+error+'%">'+value+'%</div>');
+    content.append('<div class="progress-bar progress-bar-error" style="width:'+widthArr[5]+'" title="Error: '+error+'%">'+value+'%</div>');
   }
 
-  return content
+  return content;
 
 }
 
+function getWidthProgressBar(arr){
+  var finalW = [];
+  var hash = {};
+
+  //Check if we are trying to display compliance bars on a Node Details Page.
+  try{
+    hash = JSON.parse(window.location.hash.substr(1));
+  }catch(e){};
+
+  if((Math.min.apply(Math,arr)<5)&&(hash.hasOwnProperty('nodeId'))){
+    var result = [];
+    var tamp = [];
+    var maxPercent = maxPx = 0;
+    var minPx = 22;
+    var calc;
+    $(arr).each(function(index,num){
+      if((num<5)&&(num>0)){
+        tamp = [index,num];
+        result.push(tamp);
```

Comment:
Why using result and tamp ? 
That could be done in one go:


$(arr).each(fun(index,compliance) {
  maxPercent += compliance
  maxPx += minPx 
}